ProcessLib/HT/HTProcess.cpp

```diff
@@ -229,5 +240,59 @@ void HTProcess::setCoupledSolutionsOfPreviousTimeStep()
     }
 }
 
+std::vector<double> HTProcess::getFlux(std::size_t element_id,
+                                       MathLib::Point3d const& p,
+                                       GlobalVector const& x) const
+{
+    // fetch local_x from primary variable
+    std::vector<GlobalIndexType> indices_cache;
+    auto const r_c_indices = NumLib::getRowColumnIndices(
+        element_id, *_local_to_global_index_map, indices_cache);
+    std::vector<double> local_x(x.get(r_c_indices.rows));
+
+    return _local_assemblers[element_id]->getFlux(p, local_x);
+}
+
+// this is almost a copy of the implemention in the GroundwaterFlow
+void HTProcess::postTimestepConcreteProcess(GlobalVector const& x,
+                                            const double t,
+                                            const double delta_t,
+                                            int const process_id)
+{
+    // For the monolithic scheme, process_id is always zero.
+    if (_use_monolithic_scheme && process_id != 0)
+    {
+        OGS_FATAL(
+            "The condition of process_id = 0 must be satisfied for "
+            "monolithic HTProcess, which is a single process.");
+    }
+    if (!_use_monolithic_scheme && process_id != 1)
+    {
+        DBUG("This is the thermal part of the staggered HTProcess.");
+        return;
+    }
+    if (_balance_mesh)  // computing the balance is optional
+    {
+        auto* const balance_pv = MeshLib::getOrCreateMeshProperty<double>(
+            *_balance_mesh, _balance_pv_name, MeshLib::MeshItemType::Cell, 1);
+        // initialise the PropertyVector pv with zero values
+        std::fill(balance_pv->begin(), balance_pv->end(), 0.0);
```

Comment:
Not sure here;
If the properties are set, then use NaN for initialization. If the properties are '+='-ed, then setting them 0 is of course necessary.

Reply:
The properties are  '+='-ed.